### DISCUSSION TITLE
Does websocket-client support WSS connections?

### DISCUSSION THREAD

**Discussion by CheatSense** [Marfusios/websocket-client#71] [Q&A]
Hi.

I can see in the examples that you've used the wss:// URI pattern. Using websocket-client, when I try to connect to our working secure websocket server (node ws) the client simply waits and doesn't connect, without error. 

I works perfectly in non WSS usage.

Is there something I need to do to enable this to work with secure ws servers, I don't see anything in the readme that gives me a clue what I am missing?

Thanks

**Comment by Marfusios** [comment]
Hello @TenaciousCraig ,

should be working, definitely using it in my projects. 
Check if you got valid certificate. 
Google doesn't say anything?

**Reply by CheatSense** [reply]
Hi @Marfusios 

I have Googled for ages but I have not found anything that really solves my problem. I think that maybe the certificate is the problem. Thanks!

**Reply by gmerkelvc** [reply]
Did you identify your problem?  I have a similar problem, only it works when run from an ASP.Net web application running under IIS but not when run from a console application or from a windows service.

I have a .Net 4.8 class library that uses the Marfusios Websocket-client for communicating with a secure websocket (wss) server. When running as a Windows Service or as a console application, the client times out during the initial send request after the connect. However, if the client is run within a .Net Web Application hosted by IIS 10, the client connects and communicates correctly.

When running under a Windows Service or console app on the same server, the library can connect/start the websocket client, but hangs on the Send to the server, which is triggered by the OnReconnection event for the websocket client. Using Fiddler with a proxy, I can see that the KeepAlive Ping-Pong messaging takes place after connection, but do not see the Send message that is sent in OnReconnection.

The Windows Server running the websocket client library has a valid Starfield SSL certificate installed and has TLS1.2 installed. The websocket server owner says that they are using TLS1.2, confirmed using WireShark.

Any help would be appreciated!
